Makefile

```diff
@@ -13,6 +13,7 @@ DIALECTS_BUILD_DIR ?= $(MK_DIR)/mlir/build
 RT_BUILD_DIR ?= $(MK_DIR)/runtime/build
 ENZYME_BUILD_DIR ?= $(MK_DIR)/mlir/Enzyme/build
 COVERAGE_REPORT ?= term-missing
+ENABLE_OPENQASM?=ON
```

Comment:
This seems out of place here, it's not used in this file and none of the other device options are declared here either.

Reply:
Hmm.. this is used in a few lines below to set `ASAN_OPTIONS`:

```
ifeq ($(ENABLE_OPENQASM), ON)
# A global 'mutex' is added to protect `pybind11::exec` calls concurrently in `OpenQasmRunner`.
# TODO: Remove this global 'mutex' to enable concurrent execution of remote calls.
# This 'mutex' leads to an ODR violation when using ASAN
ASAN_OPTIONS := ASAN_OPTIONS="detect_leaks=0,detect_container_overflow=0,detect_odr_violation=0"
endif
```

I had to add this here to ensure that this variable is defined and set to ON by default in this makefile.

Reply:
Oh you're right, didn't see it because it wasn't part of the changes.

It does raise a different issue though in that the asan flags are not now always set to what's required to run openqasm tests 🤔

Reply:
We could try and set the `ASAN_COMMAND` at a more granular level but it's probably not worth it. In this case we only add the `detect_odr_violation=0` flag which is not too important.